roles/agent/tasks/main.yml

```
@@ -2,6 +2,14 @@
 - name: "Include Derivate specific Variables."
   ansible.builtin.include_vars: "{{ ansible_os_family }}.yml"
 
+- name: "Get RPM or APT package facts."
+  ansible.builtin.package_facts:
+    manager: "auto"
+
+- name: "Import Legacy agent tasks."
+  ansible.builtin.include_tasks: "legacy.yml"
+  when: ansible_facts.packages['systemd'][0]['version'] | regex_search('\d{3,}') | int < 220 and checmmk_agent_prep_legacy | bool
```

Referring to my earlier comment, this task should fail, if the systemd version is unsupported and `checmmk_agent_prep_legacy` is set to `false`.

---

Do you mean by this that it should just hard fail and prevent the playbook from preceding? I imagine if you have it set to false you'll be using your own inet or have already configured it manually, I think still doing the installation would make sense. Maybe it could output a message that further configuration might be required for that system?

---

Yes, you are making sense here. Setting `checkmk_agent_prep_legacy: false` will always skip the installation, so we are good I think.